Refactor: Consolidate NLU Model

Themba-Gqaza opened the pull request:
**Description**

This PR completes a major refactor of the intent classifier. The previous collection of scripts and models has been consolidated into a single, robust, and production-ready service with a clear and stable API.

This unblocks the engineering team to begin integration and deployment work in parallel with ongoing model fine-tuning.

**What's New for Engineering:**

1. Stable, Specific API Endpoints: The old, generic /nlu/ endpoint has been replaced by two specific, production-ready endpoints as per the project spec:
    - GET /nlu/babyloss/
    -  GET /nlu/feedback/
    
2. Richer API Responses: The responses for these endpoints are now more informative. In addition to the primary answer (e.g., `babyloss: true`), the API now returns key metadata like `model_version`, `probability`, and `review_status`, which can be used for logging, monitoring, and downstream processing. The structure of these API responses is now stable and can be considered the contract for integration.

3. Standardized Deployment:
    - The service is a standard Flask/Gunicorn application. The Makefile provides a production-style make serve command and a make serve-dev command for local macOS development.
    - Critical: The trained model is now packaged as a set of artifacts in the src/artifacts/ directory. This directory must be included in the final Docker image for the service to function.

**The Handoff & Next Steps:**

While I am still fine-tuning the model to improve its accuracy, the service's architecture and API contract are now final.

This means the engineering team can proceed with:
 - Updating any services that call this NLU model to use the new endpoints.
 - Configuring the production deployment pipeline.

Future model improvements will only involve swapping out the files in the `src/artifacts` directory, requiring no changes to the application code or the API contract.

erikh360 commented on tests/test_dataset_consistency.py:
```diff
@@ -0,0 +1,100 @@
+
+
+def test_yaml_and_jsonl_counts_and_labels_match() -> None:
+    pairs = _pick_existing_pairs()
```
For some of these tests a docstring might help others understand what we are testing here.

erikh360 commented on Makefile:
```diff
@@ -0,0 +1,58 @@
+	@echo "make clean        - remove build artifacts"
+
+install:
+	$(UV) pip install -r requirements.txt
```
There is no requirements file, we use poetry for this project

erikh360 commented on pyproject.toml:
```diff
@@ -62,3 +76,24 @@ ignore = [
 
 ]
+
+# --- NEW SECTION FOR MYPY CONFIGURATION ---
```
We can probably remove this comment

erikh360 reviewed (commented):
Few smallish comments but looks good overall.

Themba-Gqaza commented:
@erikh360 I've addressed the comments you left.  Not sure how to solve the `CI / build-test-tran` but I suspect its due its because the current version is no longer aligned with the prior one
